scour/scour.py

```diff
@@ -1800,9 +1800,21 @@ def repairStyle(node, options):
                 del styleMap['overflow']
                 num += 1
 
-        # now if any of the properties match known SVG attributes we prefer attributes
-        # over style so emit them and remove them from the style map
-        if options.style_to_xml:
+    if node.nodeType == Node.ELEMENT_NODE:
+        if options.style_type == "inline":
+            # Prefer inline style
+            # Remove known SVG attributes and store their values in style attribute
+            attributes = [node.attributes.item(i).nodeName for i in range(node.attributes.length)]
+            for attribute in attributes:
+                if attribute in svgAttributes:
+                    styleMap[attribute] = node.getAttribute(attribute)
+                    node.removeAttribute(attribute)
+        elif options.style_type == "preserve":
+            # Keep whatever style of attribute versus style the file currently has
+            pass
+        elif options.style_to_xml or options.style_type == "attributes":
```

Comment:
I think it would make better sense to remove `options.style_to_xml` from here and have the initial option parsing do the relevant shuffling of values.